test/mock_query_test.dart

```diff
@@ -196,4 +196,57 @@ void main() {
       expect(snapshot.documents.first.data['tag'], equals('mostrecent'));
     }));
   });
+
+  test('Collection reference should not hold query result', () async {
+    final instance = MockFirestoreInstance();
+
+    final collectionReference = instance.collection('users');
+    await collectionReference.add({
+      'username': 'Bob',
+    });
+    final snapshot = await collectionReference.getDocuments();
+    expect(snapshot.documents, hasLength(1));
+  });
```

Comment:
This test case is the very motivation of this PR. It was failing before this PR.